mockserver-tests/src/test/groovy/pl/touk/mockserver/tests/MockServerMaxUsesTest.groovy

```diff
@@ -0,0 +1,192 @@
+            CloseableHttpResponse response2 = client.execute(restPost)
+        then:'second mock should be returned'
+            response2.entity.content.text == 'mock2'
+        when:'we call the third time using the same request'
+            CloseableHttpResponse response3 = client.execute(restPost)
+        then:'no mock should be found'
+            response3.statusLine.statusCode == 404
+    }
+
+    def 'should return two mocks in cyclic order'() {
+        given:'mock with predicate is given but for only one use'
+            remoteMockServer.addMock(new AddMock(
+                name: 'mock1',
+                path: 'testEndpoint',
+                port: 9999,
+                predicate: '''{req -> req.xml.name() == 'request'}''',
+                response: '''{req -> 'mock1'}''',
+                maxUses: 1,
+                cyclic: true
+            ))
+        and:'mock with the same predicate is given'
+            remoteMockServer.addMock(new AddMock(
+                name: 'mock2',
+                path: 'testEndpoint',
+                port: 9999,
+                predicate: '''{req -> req.xml.name() == 'request'}''',
+                response: '''{req -> 'mock2'}''',
+                maxUses: 1,
+                cyclic: true
+            ))
+        when:'we call the first time'
+            HttpPost restPost = new HttpPost('http://localhost:9999/testEndpoint')
+            restPost.entity = new StringEntity('<request/>', ContentType.create("text/xml", "UTF-8"))
+            CloseableHttpResponse response = client.execute(restPost)
+        then:'first mock should be returned and expired'
+            response.entity.content.text == 'mock1'
+        when:'we call the second time using the same request'
+            CloseableHttpResponse response2 = client.execute(restPost)
+        then:'second mock should be returned and expired'
+            response2.entity.content.text == 'mock2'
+        when:'we call the third time using the same request'
+            CloseableHttpResponse response3 = client.execute(restPost)
+        then:'first mock should be returned, because these mocks are cyclic'
+            response3.entity.content.text == 'mock1'
+        when:'we call the fourth time using the same request'
+            CloseableHttpResponse response4 = client.execute(restPost)
+        then:'second mock should be returned, because these mocks are cyclic'
+            response4.entity.content.text == 'mock2'
+    }
+
+    def 'should return two mocks with the same request interjected by another'() {
+        given:'mock with predicate is given but for only one use'
+            remoteMockServer.addMock(new AddMock(
+                name: 'mock1',
+                path: 'testEndpoint',
+                port: 9999,
+                predicate: '''{req -> req.xml.name() == 'request'}''',
+                response: '''{req -> 'mock1'}''',
+                maxUses: 1,
+                cyclic: true
```

Comment:
add test if history is not lost for cyclic mocks

Reply:
Done.